scripts/add_asimp_footer.py

```
@@ -0,0 +1,146 @@
+    if not os.path.isfile(resolved_path):
+        print(f"Skipping non-regular file: {filepath}")
+        return
+
+    # Check repository containment using commonpath
+    repo_root = os.path.realpath(os.getcwd())
+    try:
+        common = os.path.commonpath([repo_root, resolved_path])
+        if os.path.realpath(common) != repo_root:
+            print(f"Skipping file outside repository root: {filepath}")
+            return
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)